### ISSUE TITLE
CustomArgumentParser interface should provide the CommandSender caller

### ISSUE THREAD

**Issue by toddharrison** [JorelAli/CommandAPI#186]
**Please describe your suggestion**
CustomArgumentParser should provide the CommandSender so that CustomArgument implementations can use the sender to determine if there is an error.

_Example:_
Say I define a custom argument implementation where I want to allow all offline players but not the player making the call:
```kotlin
fun offlinePlayerNotSelfArgument(nodeName: String?): Argument =
    CustomArgument(nodeName) { name ->
        Bukkit.getOfflinePlayerIfCached(name)
            ?: Bukkit.getPlayer(name)
            ?: throw CustomArgumentException(MessageBuilder("That player does not exist"))
    }.overrideSuggestions { sender ->
        (Bukkit.getOnlinePlayers().map { it.name }.toSet()
                + Bukkit.getOfflinePlayers().map { it.name }.toSet())
            .filter { it != sender.name }
            .toTypedArray()
    }
```
The autocomplete list does not include the current player, but you can type in the current player and it completes. Would like to receive the sender in the CustomArgument with the node value so that I can throw an exception if the player selected is the sender.

**Comment by toddharrison** [comment]
In experimenting with this, I have confirmed that using the PlayerArgument doesn't work for offline players. I had to create custom arguments to handle offline players, and offline players excluding self.

**Comment by JorelAli** [comment]
The PlayerArgument has been rewritten in 6.0.0 (Commit 33421b59e0ea3b49b368960b299930f6ffffb1f2) to support `OfflinePlayer` objects instead of `Player` objects

**Comment by JorelAli** [comment]
Implemented in 6.0.0. OfflinePlayers can be handled using the new `OfflinePlayerArgument` and I've added an overloaded method (temporarily, I'll redesign the custom argument at some point wayyyy in the future) which lets you use a command sender.

**Comment by toddharrison** [comment]
nice work! I'll try it out once Paper bugs are fixed and I can run CommandAPI again!
